arelle/ModelXbrl.py

```diff
@@ -696,6 +696,8 @@ def createContext(
 
         XmlValidate.validate(self, newCntxElt)
         self.modelDocument.contextDiscover(newCntxElt)
+        if hasattr(self, "_dimensionsInUse"):
```

Comment:
Good catch! It would be better to only do this if the context adds new dimensions. Or even better update the set (if it's defined) with the new dimensions.

Reply:
Ok, I will change that, adding the missing dimensions should be possible

Reply:
Done!